### PR TITLE
Bot that chases the player if sees one

### PR DESCRIPTION
**Pull request by ghost**
When the bot sees the player it chases him otherwise it patrols assigned points.
#5

### CODE REVIEW

File: Source/Prot/AI/BotAIC.cpp [hunk 1]
```diff
@@ -53,10 +64,27 @@ ATargetPoint* ABotAIC::GetTargetPointByIndex(int index) const
...
 	if (Bot)
 	{
 		return Bot->PatrolPoints.Num();
 	}
 	return 0;
 }
+
+void ABotAIC::OnBotUnSee()
+{
+	GEngine->AddOnScreenDebugMessage(-1, 5.f, FColor::Green, TEXT("Can't see you!"));
```
Comment (UnoYakshi) [inline Source/Prot/AI/BotAIC.cpp]: Wrap with `if (DEBUG) {..}`.

File: Source/Prot/AI/BotAIC.cpp [hunk 2]
```diff
@@ -53,10 +64,27 @@ ATargetPoint* ABotAIC::GetTargetPointByIndex(int index) const
...
+}
+
+void ABotAIC::OnBotSee(APawn* SeenPawn)
+{
+	UWorld* World = GetWorld();
+	if (World) 
+	{
+		World->GetTimerManager().SetTimer(SeenTimerHandle, this, &ABotAIC::OnBotUnSee,
+			SensingInterval + 0.1f, false);
+		GEngine->AddOnScreenDebugMessage(-1, 5.f, FColor::Green, TEXT("I can see you!"));
```
Comment (UnoYakshi) [inline Source/Prot/AI/BotAIC.cpp]: Wrap with `if (DEBUG) {..}`.

File: enc_temp_folder/9740db6294d303de33bf82cd29a78d/BotAIC.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	//Get the possessed Pawn and check if it's Bot
+	Bot = Cast<ABot>(Pawn);
+	if (Bot)
+	{
+		//If the blackboard is valid initialize the blackboard for the corresponding behavior tree
+		if (Bot->BehaviorTree->BlackboardAsset)
+		{
+			BlackboardComponent->InitializeBlackboard(*(Bot->BehaviorTree->BlackboardAsset));
+			BlackboardComponent->SetValueAsInt("PointIndex", 0);
+			BlackboardComponent->SetValueAsBool("False", false);
```
Comment (UnoYakshi) [inline enc_temp_folder/9740db6294d303de33bf82cd29a78d/BotAIC.cpp]: What is that?

File: enc_temp_folder/9740db6294d303de33bf82cd29a78d/BotAIC.cpp [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	if (Bot)
+	{
+		return Bot->PatrolPoints.Num();
+	}
+	return 0;
+}
+
+void ABotAIC::OnBotUnSee()
+{
+	GEngine->AddOnScreenDebugMessage(-1, 5.f, FColor::Green, TEXT("Can't see you!"));
```
Comment (UnoYakshi) [inline enc_temp_folder/9740db6294d303de33bf82cd29a78d/BotAIC.cpp]: Wrap with `if (DEBUG) {..}`.

File: enc_temp_folder/9740db6294d303de33bf82cd29a78d/BotAIC.cpp [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+void ABotAIC::OnBotSee(APawn* SeenPawn)
+{
+	UWorld* World = GetWorld();
+	if (World) 
+	{
+		World->GetTimerManager().SetTimer(SeenTimerHandle, this, &ABotAIC::OnBotUnSee,
+			SensingInterval + 0.1f, false);
+		GEngine->AddOnScreenDebugMessage(-1, 5.f, FColor::Green, TEXT("I can see you!"));
```
Comment (UnoYakshi) [inline enc_temp_folder/9740db6294d303de33bf82cd29a78d/BotAIC.cpp]: Wrap with `if (DEBUG) {..}`.

File: Source/Prot/AI/BTTargetPointSelection.cpp [hunk 6]
```diff
@@ -16,10 +16,22 @@ EBTNodeResult::Type UBTTargetPointSelection::ExecuteTask(UBehaviorTreeComponent
...
+				RandomIndex = FMath::RandRange(0, AIController->GetTargetPointsNumber() - 1);
+			} while (PointIndex == RandomIndex);
+			PointIndex = RandomIndex;
+		}
+		else
+		{
+			PointIndex++;
+			if (PointIndex > AIController->GetTargetPointsNumber() - 1)
+			{
+				PointIndex = 0;
```
Comment (andrew-it) [inline Source/Prot/AI/BTTargetPointSelection.cpp:33]: Check for negative indexes

File: Source/Prot/AI/BotAIC.cpp [hunk 7]
```diff
@@ -9,6 +9,9 @@
 #include "BehaviorTree/BlackboardComponent.h"
 #include "Kismet/GameplayStatics.h"
 #include "Engine/TargetPoint.h"
+#include "Perception/PawnSensingComponent.h"
+#include "GameFramework/Pawn.h"
+#include "Engine/Engine.h"
```
Comment (UnoYakshi) [inline Source/Prot/AI/BotAIC.cpp]: Group includes by categories.

**Review by UnoYakshi — CHANGES_REQUESTED** [review]

File: Source/Prot/AI/BotAIC.cpp [hunk 8]
```diff
@@ -53,10 +64,36 @@ ATargetPoint* ABotAIC::GetTargetPointByIndex(int index) const
...
+}
+
+void ABotAIC::OnBotSee(APawn* SeenPawn)
+{
+	UWorld* World = GetWorld();
+	if (World) 
+	{
+		World->GetTimerManager().SetTimer(SeenTimerHandle, this, &ABotAIC::OnBotUnSee,
+			SensingInterval + 0.1f, false);
+		GEngine->AddOnScreenDebugMessage(-1, 5.f, FColor::Green, TEXT("I can see you!"));
```
Comment (andrew-it) [inline Source/Prot/AI/BotAIC.cpp]: Maybe `FColor::Red` ? Or another one?

**Review by UnoYakshi — APPROVED** [review]

**Pull request merged**